src/diff-snapshot.js

```diff
@@ -1,86 +1,73 @@
 /*
  * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
  *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not use this file except
  * in compliance with the License. You may obtain a copy of the License at
  *
  * http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
 
-const fs = require('fs');
-const BlinkDiff = require('blink-diff');
-const intersection = require('lodash/intersection');
+const blinkDiffComparator = require('./comparators/blink-diff');
+const pixelmatchComparator = require('./comparators/pixelmatch');
 const mkdirp = require('mkdirp');
 const path = require('path');
-
-const unsupportedDiffConfigKeys = [
-  'imageAPath',
-  'imageA',
-  'imageBPath',
-  'imageB',
-  'imageOutputPath',
-];
-
-function isDiffConfigValid(customDiffConfig) {
-  return intersection(unsupportedDiffConfigKeys, Object.keys(customDiffConfig)).length === 0;
-}
+const fs = require('fs');
+const { ResultTypes, ComparatorResult } = require('./comparator-result');
 
 function diffImageToSnapshot(options) {
   const {
     imageData,
     snapshotIdentifier,
     snapshotsDir,
     updateSnapshot = false,
-    customDiffConfig = {},
-   } = options;
-
-  if (!isDiffConfigValid(customDiffConfig)) {
-    throw new Error(
-      `Passing in options: ${unsupportedDiffConfigKeys} via Blink-Diff configuration `
-      + 'is not supported as those option are internally used. '
-      + 'Instead pass your image data as first argument to this function!'
-    );
-  }
+    comparator = 'blink-diff',
+  } = options;
 
-  let result = {};
+  // Build a path to the snapshot then check if it exists and that
+  // we're not supposed to be updating it.
   const baselineSnapshotPath = path.join(snapshotsDir, `${snapshotIdentifier}-snap.png`);
+
   if (fs.existsSync(baselineSnapshotPath) && !updateSnapshot) {
-    const outputDir = path.join(snapshotsDir, '__diff_output__');
-    const diffOutputPath = path.join(outputDir, `${snapshotIdentifier}-diff.png`);
-    const defaultBlinkDiffConfig = {
-      imageA: imageData,
-      imageBPath: baselineSnapshotPath,
-      thresholdType: 'percent',
-      threshold: 0.01,
-      imageOutputPath: diffOutputPath,
-    };
+    const comparatorOptions = Object.assign({}, options);
+
+    // Build output paths here in a single place and create necessary directories
+    comparatorOptions.baselineSnapshotPath = baselineSnapshotPath;
```

Comment:
Why not just add these to the object when defining it? I don't like the idea of modifying a `const`.